### PR TITLE
Minor UriUtils fixes

### PR DESCRIPTION
**Pull request by kriscon-db**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

- Updated the UriUtil cache to be more tolerant of property changes
- Removed existing S3 calls (new hadoopfs calls coming)
- Changed all cloud based update calls for location creation to skip for v0.2.  Only models are using this pathway in v0.2 and MLflow uses the native cloud libraries which does not require consistent path creation on the clouds.
- Removed credential empty check since we don't need credentials for v0.2 in this util.
- Pinned duckdb to 1.0.0 to unbreak the integration test

### CODE REVIEW

**Review by vikrantpuppala — APPROVED** [review]

**Comment by vikrantpuppala** [comment]
@rameshchandra could you take a look and merge?

File: .github/workflows/tutorial-tests.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
         python -m venv venv
         source venv/bin/activate
         pip install requests
-        pip install duckdb
+        pip install duckdb==1.0.0
```
Comment (dennyglee) [inline .github/workflows/tutorial-tests.yml:34]: Can you also update the instructions in the [README.md](https://github.com/unitycatalog/unitycatalog/blob/main/README.md#operate-on-delta-tables-with-duckdb) to pin `duckdb==1.0.0` (vs. the latest)?

Comment (kriscon-db) [reply .github/workflows/tutorial-tests.yml:34]: Done (the readme just said in words to use >=1.0.0, and I changed it to specify 1.0.0).

File: server/src/main/java/io/unitycatalog/server/persist/utils/UriUtils.java [hunk 2]
```diff
@@ -40,17 +36,25 @@ private enum Operation {
...
 
   // Model specific storage root handlers and convenience methods
   private static String getModelStorageRoot() {
+    String currentModelStorageRoot = properties.getProperty("storage-root.models");
+    if (modelStorageRootPropertyCached != currentModelStorageRoot) {
+      // This means the property has been updated from the previous read, or this is the first time
+      // reading it
+      reset();
+    }
     if (modelStorageRootCached != null) {
```
Comment (ravivj-db) [inline server/src/main/java/io/unitycatalog/server/persist/utils/UriUtils.java:54]: so we can't switch from project root directory to a storage root specified via properties ?

**Review by ravivj-db — APPROVED** [review]
LGTM

**Pull request merged**